debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/snapshot/MongoDbIncrementalSnapshotChangeEventSource.java

```diff
@@ -638,6 +640,19 @@ private Object[] keyFromRow(Object[] row) {
             case STRING:
                 key = documentId.asString().getValue();
                 break;
+            case BINARY:
+                var subtype = documentId.asBinary().getType();
+                if (!BsonBinarySubType.isUuid(subtype)) {
+                    throw new IllegalStateException("Unsupported type of document id");
```

Comment:
@jcechace Would it be possible to handle unrecognized binary as either `byte[]` or String-encoded byte value?

Reply:
In theory yes, however I'm not sure how well that's gonna play with our snapshotting logic (e.g. determining the max/min key values and so on).

Theoretically we could even consider ignoring UUID and just treat everything as binary

Reply:
Fair point, we cannot make any assumptions around orderin so it is better to handle it case by case.